Kernel/SanCov.cpp

```diff
@@ -6,9 +6,30 @@
 
 #include <Kernel/Arch/Processor.h>
 #include <Kernel/Devices/KCOVDevice.h>
+#include <Kernel/Library/Panic.h>
 
 extern bool g_in_early_boot;
 
+#ifdef ENABLE_KERNEL_COVERAGE_COLLECTION_DEBUG
+// Set kcov_emergency_off=true before making calls from __sanitizer_cov_trace_pc
+// to kcov instrumented code to prevent an infinite recursion. We don't want to
+// stop instrumenting all the code transitively called code used by
+// __sanitizer_cov_trace_pc during error handling. Not instrumented means we can't
+// fuzz it meaningfully.
+bool kcov_emergency_off { false };
+
+[[noreturn]] void crash_and_burn(Thread* thread);
+void crash_and_burn(Thread* thread) {
+    kcov_emergency_off = true;
+    thread->print_backtrace();
+    PANIC("KCOV is b0rked.");
+    VERIFY_NOT_REACHED();
+}
+
+#define RETURN { thread->m_kcov_recursion_hint = false; return; };
```

Comment:
This can be done by 

```
#if ENABLE_KERNEL_COVERAGE_COLLECTION_DEBUG
ScopeGuard disarm_recursion_hint = [thread] { thread->m_kcov_recursion_hint = false; };
#endif
```
Or, by using a `TemporaryChange`.

So, either AK/ScopeGuard, or AK/TemporaryChange instead of a macro.

Reply:
(Looking at the code below, using a TemporaryChange on line 60 looks better)

Reply:
Marked as TemporaryChange